### DISCUSSION TITLE
Keine Verbindung zur Datenbank

### DISCUSSION THREAD

**Discussion by Praktikant81** [shyim/shopware#81] [Q&A] [answered]
Hi zusammen,

ich habe ein Problem. Ich habe einen Docker mit Shopware am laufen.

Genutzt wird ein Container mit MySql 5.4 und einer mit Shopware.
Ich habe heute das System neu gestartet und jetzt geht nichts mehr.

Wenn ich in die Logs von Shopware schaue kommt hier
„Waiting for database connection…“

Wenn ich auf die Console der Datenbank schaue kann ich mich mit den Passwörtern anmelden.
Auch wenn ich mysql-client installiere geht die Anmeldung.

Auch eine komplette saubere Installation bringt keinen Erfolg.

Hat jemand eine Idee was ich flasch mache bzw. was ich machen kann?

Danke schon mal im Voraus.

Gruß Dirk

**Comment by Praktikant81** [comment]
sorry gerade gesehen meinte natürlich MySql 5.7

**Comment by shyim** [comment]
Welches genaue Image benutzst du gerade?

**Comment by Praktikant81** [comment, marked as answer]
Hey danke für deine Rückmeldung.

Ich habe das Problem gefunden.
In der Docker-Compose war DATABASE_PORT nicht gesetzt was die ganze Zeit ohne Probleme funktioniert hat. Leider nach dem Neustart dann halt nicht mehr.
Nach einfügen und setzen des Ports ist der Container wieder normal gestartet.

Gruß Dirk

**Reply by shyim** [reply]
Wenn du das docker image neu pullst sollte es gehen

**Reply by Praktikant81** [reply]
also irgendwie möchte es jetzt nicht mehr.

Ich habe neu gepullt jetzt geht die Verbingung zu Mysql auch wieder ohne den Eintrag aber der Shop geht nicht.
Ich habe ab der Docker-Compose nichts geändert.

Das Image ist: shyim/shopware:latest

Im Portainer wird der Container mit unhealthy angezeigt.

Also wenn ich jetzt den Container starte bekomme ich im Browser die Meldung "File not found"

Im Docker Log kommen diese Meldungen

Waiting for database connection...
mysql (172.26.0.3:3306) open
time="2023-04-13T19:08:47Z" level=info msg="load configuration from file" file=/etc/supervisord.conf
time="2023-04-13T19:08:47Z" level=info msg="create process:php-fpm"
time="2023-04-13T19:08:47Z" level=info msg="create process:nginx"
time="2023-04-13T19:08:47Z" level=info msg="create process:shopware-empty-cache"
time="2023-04-13T19:08:47Z" level=info msg="stop listening"
time="2023-04-13T19:08:47Z" level=info msg="try to start program" program=php-fpm
time="2023-04-13T19:08:47Z" level=info msg="try to start program" program=nginx
time="2023-04-13T19:08:47Z" level=info msg="try to start program" program=shopware-empty-cache
time="2023-04-13T19:08:47Z" level=info msg="success to start program" program=shopware-empty-cache
time="2023-04-13T19:08:47Z" level=info msg="program stopped with status:exit status 0" program=shopware-empty-cache
2023/04/13 19:08:47 [notice] 22#22: using the "epoll" event method
2023/04/13 19:08:47 [notice] 22#22: nginx/1.22.1
2023/04/13 19:08:47 [notice] 22#22: OS: Linux 5.10.0-21-amd64
2023/04/13 19:08:47 [notice] 22#22: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2023/04/13 19:08:47 [notice] 22#22: start worker processes
2023/04/13 19:08:47 [notice] 22#22: start worker process 24
2023/04/13 19:08:47 [notice] 22#22: start worker process 25
2023/04/13 19:08:47 [notice] 22#22: start worker process 26
2023/04/13 19:08:47 [notice] 22#22: start worker process 27
[13-Apr-2023 19:08:47] NOTICE: fpm is running, pid 23
[13-Apr-2023 19:08:47] NOTICE: ready to handle connections
time="2023-04-13T19:08:48Z" level=info msg="success to start program" program=php-fpm
time="2023-04-13T19:08:48Z" level=info msg="success to start program" program=nginx
127.0.0.1 -  13/Apr/2023:19:09:09 +0000 "GET /index.php" 404
192.168.178.68 - - [13/Apr/2023:19:09:09 +0000] "GET / HTTP/1.1" 404 47 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/112.0"
2023/04/13 19:09:09 [info] 24#24: *1 client 192.168.178.68 closed keepalive connection
127.0.0.1 -  13/Apr/2023:19:09:12 +0000 "GET /index.php" 404
127.0.0.1 - - [13/Apr/2023:19:09:12 +0000] "GET /admin HTTP/1.1" 404 27 "-" "curl/7.83.1"
2023/04/13 19:09:12 [info] 26#26: *3 client 127.0.0.1 closed keepalive connection
